src/Chainweb/Pact/Service/PactInProcApi.hs

```diff
@@ -89,15 +89,34 @@ withPactService'
     -> (PactQueue -> IO a)
     -> IO a
 withPactService' ver cid logger memPoolAccess bhDb pdb sqlenv config action = do
-    reqQ <- atomically $ newPactQueue (_pactQueueSize config)
-    race (server reqQ) (client reqQ) >>= \case
+    reqQ <- newPactQueue (_pactQueueSize config)
+    race (concurrently_ (monitor reqQ) (server reqQ)) (client reqQ) >>= \case
         Left () -> error "pact service terminated unexpectedly"
```

Comment:
Please add a `HasCallStack` constraint to this function, which would help to trace the error in case it triggers. Also, please prefix the error message with the Module and function name.